juju/juju
# [JUJU-521] 1959351 fix slow juju ssh leader

Find the leader for `juju ssh application/leader` via a direction api call rather than gathering the data from status to improve result time.  Fall back to the current method if the new api call is not available.

Including a little bit of cleanup due to compiler warnings etc.

## QA steps

Deploy a bunch of units and time juju ssh to the leader of the application.  It should be faster with the newer client and controller.  

Ensure that the newer client can still run juju ssh to the leader of the application, though slower.

Test with debug hooks and in a k8s config as well.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1959351
